File: www/JLRecorder.js
/*
 * File: JLRecorder.js
 * Project: com.undsky.cordova.jlrecorder
 * File Created: 2018-09-28 08:22:19 am
 * Author: 姜彦汐 (jiangyanxi@live.com)
 * -----
 * Last Modified: 2018-09-28 09:46:15 am
 * Modified By: 姜彦汐 (jiangyanxi@live.com>)
 * -----
 * Copyright (c) 2018 www.undsky.com
 */

var exec = require('cordova/exec');

module.exports = {
    /**
     *录音
     *
     * @param {*} path mp3 存储路径
     * @param {*} success
     * @param {*} error
     */
    start: function (path, success, error) {
        exec(success, error, 'JLRecorder', 'start', [path]);
    },
    /**
     *暂停
     *
     * @param {*} success
     * @param {*} error
     */
    stop: function (success, error) {
        exec(success, error, 'JLRecorder', 'stop', []);
    },
    /**
     *销毁
     *
     * @param {*} success
     * @param {*} error
     */
    destroy: function (success, error) {
        exec(success, error, 'JLRecorder', 'destroy', []);
    },
};

File: src/ios/JLRecorder.m
/*
 * File: JLRecorder.m
 * Project: com.undsky.cordova.jlrecorder
 * File Created: 2018-09-28 08:51:09 am
 * Author: 姜彦汐 (jiangyanxi@live.com)
 * -----
 * Last Modified: 2018-09-28 09:46:53 am
 * Modified By: 姜彦汐 (jiangyanxi@live.com>)
 * -----
 * Copyright (c) 2018 www.undsky.com
 */

#import <Cordova/CDV.h>

@interface JLRecorder : CDVPlugin {
  // Member variables go here.
}

- (void)start:(CDVInvokedUrlCommand*)command;
- (void)stop:(CDVInvokedUrlCommand*)command;
- (void)destroy:(CDVInvokedUrlCommand*)command;
@end

@implementation JLRecorder

-(void)pluginInitialize{

}

- (void)start:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* path = [command.arguments objectAtIndex:0];

    if (path != nil && [path length] > 0) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:path];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:"need path argument"];
    }

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)stop:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)destroy:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];

    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end
